### ISSUE TITLE
Breaking change in 1.2.3 - Inaccessible directory if file that starts with the same name as the directory is also present.

### ISSUE THREAD

**Issue by tonybendis** [Azure/azure-storage-fuse#412]
### Which version of the blobfuse was used?
1.2.3

### Which OS (please include version) are you using?
Ubuntu 16.04

### What problem was encountered?
Inaccessible directory if file that starts with the same name as the directory is also present.

### Have you found a mitigation/solution?
Downgrade to version 1.2.1

### Problem details:
I have the following blobs in a blob container:

mydir1/testfile.txt
mydir2/testfile.txt
mydir3/testfile.txt
mydir1.txt
mydir2blah
mydir.txt

With blobfuse 1.2.3, presence of mydir1.txt and mydir2blah makes mydir1 and mydir2 inaccessible. Works fine with blobuse 1.2.1.
Maybe the recent change in line 440 in utilities.cpp?

With blobfuse 1.2.3:

ls -la
d????????? ? ?    ?       ?            ? mydir1
-rwxrwxrwx 1 root root    4 Jun  2 19:33 mydir1.txt
d????????? ? ?    ?       ?            ? mydir2
-rwxrwxrwx 1 root root    4 Jun  2 19:33 mydir2blah
drwxrwxrwx 2 root root 4096 Jan  1  1970 mydir3
-rwxrwxrwx 1 root root    4 Jun  2 19:33 mydir.txt

cd mydir1
bash: cd: mydir1: No such file or directory

cd mydir3
(success)

With blobfuse 1.2.1:

ls -la
drwxrwxrwx 3 root root 4096 Jan  1  1970 mydir1
-rwxrwxrwx 1 root root    4 Jun  2 19:33 mydir1.txt
drwxrwxrwx 3 root root 4096 Jan  1  1970 mydir2
-rwxrwxrwx 1 root root    4 Jun  2 19:33 mydir2blah
drwxrwxrwx 3 root root 4096 Jan  1  1970 mydir3
-rwxrwxrwx 1 root root    4 Jun  2 19:33 mydir.txt

cd mydir1
(success)

**Comment by andyzhangx** [comment]
@NaraVen do you know how can I install 1.2.1 in Ubuntu? `apt install blobfuse` will always install 1.2.3 which has broken change.  We are trying to install a fixed version on Ubuntu to prevent such breaking change in the future. Thanks.
```console
# apt show blobfuse
Package: blobfuse
Version: 1.2.3
Priority: optional
Section: devel
Maintainer: Microsoft - Azure Storage
Installed-Size: 26.9 MB
Depends: fuse
Download-Size: 8,640 kB
APT-Sources: https://packages.microsoft.com/ubuntu/16.04/prod xenial/main amd64 Packages
Description: blobfuse 1.2.3 - FUSE adapter for Azure Blob Storage
```

**Comment by NaraVen** [comment]
This bug is fixed already so I don't know why you want to take an earlier version.
However if that's what you want to do here are the steps,
List them 
sudo apt list -a blobfuse
pick the version you want
sudo apt-get install blobfuse=1.2.1 ( or whatever version you want)

**Comment by andyzhangx** [comment]
> This bug is fixed already so I don't know why you want to take an earlier version.
> However if that's what you want to do here are the steps,
> List them
> sudo apt list -a blobfuse
> pick the version you want
> sudo apt-get install blobfuse=1.2.1 ( or whatever version you want)

@NaraVen so 1.2.3 already has this fix: https://github.com/Azure/azure-storage-fuse/pull/413 ?

**Comment by NaraVen** [comment]
yes 1.2.3 has the fix for #413. It was released as a hot fix so we did not change the version number.

**Comment by andyzhangx** [comment]
> yes 1.2.3 has the fix for #413. It was released as a hot fix so we did not change the version number.

A few customers are worried about such broken issues since we always install latest release in our VM, that may break sth in the future, use a fixed version could fix such issue. @NaraVen

**Comment by NaraVen** [comment]
Good point, Noted. We are automating our build and release process, really close to committing it. So going forward every release will have an incremented patch number or a whole new minor version number. Also, for your convenience, in my earlier comment I have the steps for installing a fixed version.
